src/java/org/apache/cassandra/db/compaction/unified/AdaptiveController.java

```diff
@@ -204,6 +205,10 @@ static Controller fromOptions(Environment env,
         {
             logger.warn("Unable to parse saved options. Using starting value instead:", e);
         }
+        catch (Throwable e) {
+            logger.warn("Unable to read controller config file. Using starting value instead:", e);
+            JVMStabilityInspector.inspectThrowable(e);
```

Comment:
I had a closer look at the inspectors (e.g. `LenientIOErrorHandler`). It seems that some of them kill the process on `FSError`. WDYT about catching`FSError` without inspection and `Throwable` with inspection separately?